youtube_dl/extractor/youtube.py

```diff
@@ -2123,7 +2123,7 @@ def gen_owner_profile_url():
                 or search_meta('uploadDate')),
             'uploader': video_details['author'],
             'uploader_id': self._search_regex(
-                r'/(?:channel|user)/([^/?&#]+)', owner_profile_url,
+                r'/(?:(?:channel|user)/|@)([^/?&#]+)', owner_profile_url,
```

Comment:
```suggestion
                r'/(?:(?:channel/|user/|(?=@)))([^/?&#]+)', owner_profile_url,
```
The `@` is a part of the id imo. Atleast, that's how we decided to return it in https://github.com/yt-dlp/yt-dlp

Reply:
I have also unit tests changes, or else it fails. See the changes here: https://github.com/arlm/youtube-dl/tree/backport-upload_id-fix-from-yt-dlp

By the way, @pukkandan how do I run only YouTube related tests, but all of them? Many seem to be skipped and by using `python test/test_download.py TestDownload.test_Youtube` it runs just the ones I fixed.

Reply:
You can use `unittest -k`/`pytest -k`

Reply:
The PR should contain all necessary changes including updated tests, please.

Reply:
I think I'm convinced that the `uploader_id` should include the `@`, since the remainder may be a string that doesn't occur anywhere without the initial `@`. Eg, in the canonical test video, `PhilippHagemeister`